include/hpp/constraints/matrix-view.hh

```diff
@@ -873,12 +873,16 @@ class MatrixBlockView
     }
     // ++it
     block_iterator& operator++() {
-      _ro.setValue(_ro.value() + rs());
+      if (!AllRows) {
+        _ro.setValue(_ro.value() + rs());
+      }
       ++row;
       if (row == (size_type)view.m_rows.size()) {
```

Comment:
When `AllRows` is true, I would expect that row stays at `0`. An assert would make it explicit and improve readability.
```c++
assert(view.m_rows.size() == 1);
assert(row == 0); // before ++row;
```